### PR TITLE
Delete devtools_testing package

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Fixes https://github.com/flutter/devtools/issues/3172

This PR moves all the code from devtools_testing into devtools_app/test. This PR also adds flutter-sdk/ to the analysis-options exclude section and reverts the test fixtures provider_app/ and riverpod_app/ into non-null-safe mode (since devtools_app is not yet null safe and this PR moves the test fixtures into the devtools_app package).

### CODE REVIEW

**Comment by devoncarew** [comment]
This is still a WIP? Happy to review when ready.

File: tool/bots.sh [hunk 1]
```diff
@@ -95,7 +95,8 @@ if [ "$BOT" = "main" ]; then
     repo_tool repo-check
 
     # Analyze the source.
-    dart analyze --fatal-infos
+    dart analyze --fatal-infos lib
```
Comment (devoncarew) [inline tool/bots.sh:98]: duplicated line

Comment (devoncarew) [reply tool/bots.sh:98]: Also, why just analyze the source in `lib/`? Don't we want all the source in devtools_app?

If there are some sub-dirs we want to exclude we can do that w/ some analysis options surgery.

**Review by devoncarew — APPROVED** [review]
lgtm w/ one comment on the tool/bots.sh script

File: tool/bots.sh (hunk 1 above)
Comment (kenzieschmoll) [reply tool/bots.sh:98]: good catch, this change accidentally leaked in. reverting.

**Pull request merged**